ehcache-impl/src/main/java/org/ehcache/impl/persistence/DefaultLocalPersistenceService.java

```diff
@@ -244,4 +275,38 @@ public File getRoot() {
       return safeSpace.directory;
     }
   }
+
+  private void onStartupIfLockFileExists() {
```

Comment:
This method appears to be duplicating a lot of the logic already in the `internalStart()` method. It seems this could be greatly simplified if you took a more strict "lock first, then check for clean" approach?